### PR TITLE
Add Mindflayer tests to Tox integrations

### PR DESCRIPTION
**Pull request by nuk380y**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
+    lint,
+    sanity,
+    functional
 skipdist = True
 skip_missing_interpreters = True
 
 [travis]
 python =
-  3.7: lint, functional
+  3.7: lint, sanity, functional
```
Comment (ragingpastry) [inline tox.ini]: Let's remove functional until we figure out how to properly test this one

File: tox.ini [hunk 2]
```diff
@@ -38,41 +36,32 @@ sitepackages = false
...
-[testenv:doc]
-# doc requires py3 due to use of f'' strings and using only python3 as
-# basepython risks using python3.4 which is not supported.
-basepython = python3.7
-passenv = *
 commands =
-    python setup.py build_sphinx -n -W --builder=html
-extras =
-    docs
+    flake8 piedpiper-noop-function
```
Comment (ragingpastry) [inline tox.ini]: This needs to either glob "piedpiper-noop" or we need to specify "piedpiper-noop-gateway" as well

File: tox.ini [hunk 3]
```diff
@@ -38,41 +36,32 @@ sitepackages = false
...
+    pluggy==0.12.0
+    pytest==4.5.0
+    PyYAML==5.1
+skip_install = true
+usedevelop = false
 commands_pre =
-# build the docker container
+    wget https://github.com/AFCYBER-DREAM/piedpiper-mindflayer/archive/v1.0.0.zip \
+        -O /tmp/piedpiper-mindflayer.zip
+    unzip /tmp/piedpiper-mindflater.zip
```
Comment (ragingpastry) [inline tox.ini]: /mindflater/mindflayer/

File: tox.ini [hunk 4]
```diff
@@ -38,41 +36,32 @@ sitepackages = false
...
-        --pull --rm \
-        {posargs:-t quay.io/afcyber-dream/piedpiper-flake8-faas:$TAG} .\
-    '
-whitelist_externals =
-    sh
+    pytest ./piedpiper-mindflayer-1.0.0/tests/test_stack_yaml.py
+commands_post =
+    rm /tmp/piedpiper-mindflayer.zip
+
+[testenv:functional]
```
Comment (ragingpastry) [inline tox.ini]: Let's just remove this environment entirely for now

**Review by ragingpastry — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (nuk380y) [reply tox.ini]: `[testenv:functional]` has been completely removed.

File: tox.ini (hunk 2 above)
Comment (nuk380y) [reply tox.ini]: For sake of easy debugging in the future, a second `flake8` command was given in the `[testenv:lint]` checks.

File: tox.ini (hunk 3 above)
Comment (nuk380y) [reply tox.ini]: I hate typoos, too.

File: tox.ini (hunk 4 above)
Comment (nuk380y) [reply tox.ini]: See previous comment.

**Review by ragingpastry — CHANGES_REQUESTED** [review]
Sorry for all the back and forth. Last thing.

Instead of specifying the two directories specifically.. let's just run flake8 on the whole thing.

`flake8 --exclude=.tox,template`

Would make the tox file more portable

**Comment by nuk380y** [comment]
Replaced explicit-inclusive flake8 calls with explicit-exclusive flake8 call.

**Comment by nuk380y** [comment]
Forgot to put character and complexity limits on flake8 in `tox.ini`.  Added those to the pipeline as well as having Mindflayer unzip and execute from the `/tmp/` directory.

**Review by ragingpastry — APPROVED** [review]
Looks good. Lint failures are expected. Will open another PR to fix these

**Pull request merged**